src/HotChocolate/PersistedOperations/src/PersistedOperations.AzureBlobStorage/AzureBlobOperationDocumentStorage.cs

```diff
@@ -0,0 +1,118 @@
+using Azure;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using HotChocolate.Execution;
+using HotChocolate.Language;
+
+namespace HotChocolate.PersistedOperations.AzureBlobStorage;
+
+/// <summary>
+/// An implementation of <see cref="IOperationDocumentStorage"/> that uses Redis as a storage.
+/// </summary>
+public class AzureBlobOperationDocumentStorage : IOperationDocumentStorage
+{
+    private static readonly BlobOpenWriteOptions _defaultBlobOpenWriteOptions = new()
+    {
+        HttpHeaders = new BlobHttpHeaders
+        {
+            ContentType = "application/graphql",
+            ContentDisposition = "inline",
+            CacheControl = "public, max-age=604800, immutable"
+        }
+    };
+
+    private readonly BlobContainerClient _blobContainerClient;
+    private readonly string _blobNamePrefix;
+    private readonly string _blobNameSuffix;
+
+    /// <summary>
+    /// Initializes a new instance of the class.
+    /// </summary>
+    /// <param name="containerClient">The blob container client instance.</param>
+    /// <param name="blobNamePrefix">This prefix string is prepended before the hash of the document.</param>
+    /// <param name="blobNameSuffix">This suffix is appended after the hash of the document.</param>
+    public AzureBlobOperationDocumentStorage(
+        BlobContainerClient containerClient,
+        string blobNamePrefix,
+        string blobNameSuffix)
+    {
+        ArgumentNullException.ThrowIfNull(containerClient);
```

Comment:
We cannot use `ArgumentNullException.ThrowIfNull` if you want this to be ported back to 14 as 14 still supports .NET Standard 2.0.